src/common/mfu_daos.c

```diff
@@ -1095,7 +1098,11 @@ int daos_connect(
         /* Try to open the container
          * If NOEXIST we create it */
         daos_cont_info_t co_info = {0};
-        rc = daos_cont_open(*poh, cont_uuid, DAOS_COO_RW, coh, &co_info, NULL);
+        if (!dst_cont_passed && create_cont) {
+            rc = daos_cont_open(*poh, da->dst_cont_uuid, DAOS_COO_RW, coh, &co_info, NULL);
+        } else {
+            rc = daos_cont_open(*poh, *cont, DAOS_COO_RW, coh, &co_info, NULL);
```

Comment:
Would you please include a comment in these else blocks like you have around line 1163?  Since the ``if`` has two conditions ``!dst_cont_passed && create_cont``, it helps to explain why the else captures all of the rest of the cases.

Reply:
@adammoody ok, thanks i added a comment